From 1f487c42eb2be4e252e5f581d3f9f5396d38211b Mon Sep 17 00:00:00 2001
From: Weng Xuetian <wengxt@gmail.com>
Date: Mon, 14 Sep 2015 18:21:08 -0700
Subject: [PATCH] Close the pipe fd passed to X server.

If X server fails to write the display number to the pipe, sddm may
freeze on the next readLine call because it is still possible to read
from it.
---
 src/daemon/XorgDisplayServer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/daemon/XorgDisplayServer.cpp b/src/daemon/XorgDisplayServer.cpp
index b40f718..a7e8436 100644
--- a/src/daemon/XorgDisplayServer.cpp
+++ b/src/daemon/XorgDisplayServer.cpp
@@ -175,6 +175,10 @@ namespace SDDM {
                 return false;
             }
 
+            // close the other side of pipe in our process, otherwise reading
+            // from it may stuck even X server exit.
+            close(pipeFds[1]);
+
             QFile readPipe;
 
             if (!readPipe.open(pipeFds[0], QIODevice::ReadOnly)) {
-- 
1.9.0

